sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/service/storage/common/AbstractAzureStorageClientBuilderFactory.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.service.storage.common;
+
+import com.azure.spring.core.converter.AzureRequestRetryOptionsConverter;
+import com.azure.spring.core.factory.AbstractAzureHttpClientBuilderFactory;
+import com.azure.spring.core.properties.retry.RetryProperties;
+import com.azure.spring.core.properties.retry.StorageRetryProperties;
+import com.azure.storage.common.policy.RequestRetryOptions;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Abstract factory for Storage client builder.
+ * @param <T> Storage service builder
+ */
+public abstract class AbstractAzureStorageClientBuilderFactory<T> extends AbstractAzureHttpClientBuilderFactory<T> {
+
+    private final AzureRequestRetryOptionsConverter requestRetryOptionsConverter = new AzureRequestRetryOptionsConverter();
+
+    protected BiConsumer<T, RequestRetryOptions> consumeRequestRetryOptions() {
+        return (a, b) -> { };
+    }
```

Comment:
No need to be here since we have implementations?

Reply:
Or should it be abstract here?

Reply:
abstract is better.